# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatformFontList.h
# Commit: 6af07e63a001
# Full Hash: 6af07e63a001dfc667a952727551a80b16797339
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-08-13 09:54:32
# Regressor Bug: 1565966
# File Overlap Count: 1
# Description:
#   Bug 1565966 - part 2 - Include Family attributes in mAliasTable when collecting font-name aliases, so that fontEntry instantiation on Windows can work correctly for the Aliases eventually stored in the font-list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D41243
# ==============================================================================

diff -r 277876c31869 -r 6af07e63a001 gfx/thebes/gfxPlatformFontList.h
--- a/gfx/thebes/gfxPlatformFontList.h	Mon Aug 12 17:49:09 2019 +0000
+++ b/gfx/thebes/gfxPlatformFontList.h	Mon Aug 12 17:49:28 2019 +0000
@@ -27,6 +27,12 @@
 #include "mozilla/RangedArray.h"
 #include "nsLanguageAtomService.h"
 
+namespace mozilla {
+namespace fontlist {
+struct AliasData;
+}
+}  // namespace mozilla
+
 class CharMapHashKey : public PLDHashEntryHdr {
  public:
   typedef gfxCharacterMap* KeyType;
@@ -431,6 +437,8 @@
       mozilla::StyleGenericFontFamily aGenericType);
 
  protected:
+  friend class mozilla::fontlist::FontList;
+
   class InitOtherFamilyNamesRunnable : public mozilla::CancelableRunnable {
    public:
     InitOtherFamilyNamesRunnable()
@@ -778,8 +786,7 @@
 
   mozilla::UniquePtr<mozilla::fontlist::FontList> mSharedFontList;
 
-  nsClassHashtable<nsCStringHashKey, nsTArray<mozilla::fontlist::Pointer>>
-      mAliasTable;
+  nsClassHashtable<nsCStringHashKey, mozilla::fontlist::AliasData> mAliasTable;
   nsDataHashtable<nsCStringHashKey, mozilla::fontlist::LocalFaceRec::InitData>
       mLocalNameTable;
 
